QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java

```diff
@@ -581,150 +573,136 @@ void setPanelFragment(int side, Fragment fragment) {
-            if (hasDrawer && locked)
-                extensibleDrawerToggle.getDrawer().setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED, side);
-            else if (hasDrawer)
-                extensibleDrawerToggle.getDrawer().setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED, side);
+        public void cleanupMenus() {
+            nickFragment.setMenuVisibility(false);
+            chatFragment.setMenuVisibility(false);
+            detailFragment.setMenuVisibility(false);
         }
 
-        public void setWindowProperties(int side) {
-            switch (side) {
-                case Gravity.START:
-                    bufferFragment.setMenuVisibility(true);
-                    chatFragment.setMenuVisibility(false);
-                    break;
-                case Gravity.END:
-                case Gravity.NO_GRAVITY:
-                default:
-                    bufferFragment.setMenuVisibility(false);
-                    chatFragment.setMenuVisibility(true);
-                    break;
+        @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+        public void setupDrawer() {
+            leftDrawer = (DrawerLayout) findViewById(R.id.main_drawer);
+            rightDrawer = (DrawerLayout) findViewById(R.id.nick_drawer);
+
+            if (leftDrawer!=null) {
+                leftDrawer.setStatusBarBackgroundColor(getResources().getColor(R.color.primary_dark));
+
+                extensibleDrawerToggle = new ExtensibleDrawerToggle(leftDrawer, new ActionBarDrawerToggle(
+                        MainActivity.this, /* host Activity */
+                        leftDrawer, /* DrawerLayout object */
+                        actionbar.getWrappedToolbar(), /* nav drawer icon to replace 'Up' caret */
+                        R.string.hint_drawer_open, /* "open drawer" description */
+                        R.string.hint_drawer_close /* "close drawer" description */
+                ) {
+                    /**
+                     * Called when a drawer has settled in a completely closed state.
+                     */
+                    public void onDrawerClosed(View drawerView) {
+                        updateBufferRead();
+                        updateBufferRead();
+                        ((BufferFragment) manager.bufferFragment).finishActionMode();
+                        setTitleAndMenu();
+                    }
+
+                    /**
+                     * Called when a drawer has settled in a completely open state.
+                     */
+                    public void onDrawerOpened(View drawerView) {
+                        manager.closeDrawer(Side.RIGHT);
+                        setTitleAndMenu();
+                    }
+
+                    public void onDrawerSlide(View drawerView, float slideOffset) {
+                        if (manager.isDrawerVisible(Side.RIGHT))
+                            manager.closeDrawer(Side.RIGHT);
+                        super.onDrawerSlide(drawerView,0);
```

Comment:
The guidelines say that, if the drawer slides over the button, the button always should stay in position 0.